src/main/java/ca/on/oicr/gsi/dimsum/CaseLoader.java

```diff
@@ -844,4 +854,47 @@ private static Map<String, ProjectSummary> buildProjectSummaries(
     return projectSummariesByName;
   }
 
+  private static Stream<Sample> filterSampleByDate(List<Sample> samples,
+      Collection<DateFilter> filters) {
+    Stream<Sample> stream = samples.stream();
+    if (filters != null && !filters.isEmpty()) {
+      Map<DateFilterKey, Predicate<Sample>> filterMap = new HashMap<>();
+      for (DateFilter filter : filters) {
+        DateFilterKey key = filter.getKey();
+        if (filterMap.containsKey(key)) {
+          filterMap.put(key, filterMap.get(key).or(filter.samplePredicate()));
+        } else {
+          filterMap.put(key, filter.samplePredicate());
+        }
+      }
```

Comment:
It doesn't make sense to have multiple of the same date filter, so this mapping can be skipped. Each filter can be applied as is.

The results from this method are never really used - you just check if there are any matching items. You could simplify by renaming and changing this method to return a boolean instead:

```
samples.stream()
        .anyMatch(sample -> filters.stream().allMatch(filter -> filter.samplePredicate().test(sample)));
```

Can do similar for `filterRequisitionByQcGate`